### ISSUE TITLE
[BUG] npm install modifies the package.json funding field

### ISSUE THREAD

**Issue by jaydenseric** [npm/cli#1964]
### Current Behavior:

Running `npm install` with npm v7 undesirably modifies the `package.json` funding field from a URL string, to an object with a `url` property.

### Expected Behavior:

Running `npm install` should not modify [a valid format for the `package.json` funding field](https://docs.npmjs.com/configuring-npm/package-json.html#funding).

### Steps To Reproduce:

1. In `package.json`, add a `funding` field with a URL string as a value.
2. In the project, run `npm install`.

Notice the resulting change to `package.json`, e.g:

```diff
-  "funding": "https://github.com/sponsors/jaydenseric",
+  "funding": {
+    "url": "https://github.com/sponsors/jaydenseric"
+  },
```

### Environment:

- Node.js: v14.13.1
- npm: v7.0.0

**Comment by billyjanitsch** [comment]
It also modifies the `bin` field, e.g.:

```diff
 {
   "name": "foo",
-  "bin": "bar.js"
+  "bin": {
+     "foo": "bar.js"
+   }
 }
```

**Comment by Farfurix** [comment]
I have a similar issue with using relative paths in the "bin" field.
My steps (npm v7.0.5):
1. Run `npm init -y`
2. Add the "bin" field with some relative path:

```json
"bin": {
  "foo": "./bin/bar.js"
},
```

3. Run `npm i`
4. Result:

```diff
"bin": {
-   "foo": "./bin/bar.js"
+   "foo": "bin/bar.js"
},
```

**Comment by ljharb** [comment]
If anything, it should *include* the `./` - but ideally it wouldn't modify the field value at all.

**Comment by sindresorhus** [comment]
@ruyadorno Any chance you would be able to look into this? Anyone using my [`np`](https://github.com/sindresorhus/np) tool to publish packages are unable to use npm v7 because `np` runs `npm install` when publishing and since that modifies package.json, the Git dirty check fails and so the whole publish fails.

**Comment by ruyadorno** [comment]
Fixed in `7.0.15` 😊
